### ISSUE TITLE
2.1.22 - Password history gets wiped out after upgrade

### ISSUE THREAD

**Issue by hexxamillion** [nilsteampassnet/TeamPass#1026]
I am upgrading from 2.1.16 -> 2.1.22. I tested a few times and it seems that after upgrading the password history gets wiped out. Any idea on why this would happen? The only entries that show up in password history are folder modifications. Password modifications are gone.

**Comment by hexxamillion** [comment]
The same thing happens when I try to upgrade from 2.1.16 -> 2.1.23.

**Comment by hexxamillion** [comment]
I tried upgrade again and then ran a restore for the log_items table but that did not work. The password history is still missing. @nilsteampassnet  Can you provide detail on the logic of password history like what table(s) are used for it? Also during the upgrade process what is happening to those tables that might wipe out previous history?

**Comment by hexxamillion** [comment]
Racking my brains on this one. I cannot find anywhere in the database for password history entries. I tried adding new and cannot locate where it ended up in the DB. I am trying to understand how the password history is getting wiped out during upgrade from 2.1.22 but it would help to understand the complete logic around storing password history.

**Comment by nilsteampassnet** [comment]
Logically there is no wipe of any table during upgrade process. 
I need to check the différenciés between version 16 and 23 to be able to understand this issue. 
Items history is stored in take items_log 

Will keep you informed after investigations

**Comment by hexxamillion** [comment]
Awesome.  Thanks!

Regards,

Hector
(Sent from mobile)

**Comment by hexxamillion** [comment]
Another note is that I am able to add new and those show up fine. The problem is pre-existing password history does not show after upgrade. I see them in the log_items table so they are there. Could it be maybe a change in the encryption process in the new version that is preventing them from viewing?

**Comment by hexxamillion** [comment]
Any update on this one?

**Comment by nilsteampassnet** [comment]
​I cannot understand this one and could not reproduce.
There is no reason to have such a behavior.
History is not encrypted so the recent encryption changes has nothing to do.

I continu to investigate​

**Comment by nilsteampassnet** [comment]
When you say "Password modifications are gone.", do you mean that you
cannot see the previous passwords used or that you don't see any entry at
all?

----
*Nils @ teampass.net <http://teampass.net>*
*Teampass *- *A collaborative passwords manager*
Visit Teampass.net
Visit Github repository <https://github.com/nilsteampassnet/TeamPass>


2015-10-23 16:21 GMT+02:00 Nils Laumaillé <nils@teampass.net>:

> ​I cannot understand this one and could not reproduce.
> There is no reason to have such a behavior.
> History is not encrypted so the recent encryption changes has nothing to
> do.
>
> I continu to investigate​
>
>

**Comment by nilsteampassnet** [comment]
​Ok I think that the upgrade process doesn't change the encryption of the
passwords in history.
So you are right, they are not displayed in the history ... I need to
improve :-)​

**Comment by hexxamillion** [comment]
Super! That sounds about right because anything new is added to history just fine. It's just the items prior to upgrade that are not seen.

**Comment by hexxamillion** [comment]
Any update on this? For which release is this planning for. This is an important one and the reason I have not upgraded from 2.1.16 yet.

**Comment by nilsteampassnet** [comment]
​I've added some code to handle this, and in my test server it works but
perhaps my scenario is not enough complete to be representative to all
usage.
I need to go deeper inside investigation​

**Comment by nilsteampassnet** [comment]
​@hexxamillion​, I think I've fond the bug and corrected it using a 2.1.26
database.
Can you please use branch "feature_branch" in which I have committed the
fix, and give me a feedback?
Thanks

**Comment by hexxamillion** [comment]
Will do! Thanks and stay tuned!

**Comment by hexxamillion** [comment]
Installation was smooth but unfortunately I cannot log in due to the same issue as #878 . 

Uncaught exception 'DomainException' with message 'Unsupported Password Hash Supplied'. Then in the log it still has the password in clear text. I tried the suggestions again in #878  but no success. I need to get past this part and login for me to validate the password history.

**Comment by nilsteampassnet** [comment]
I don't know how ldap works and can not test it.

How could I help here?

**Comment by hexxamillion** [comment]
Not sure. I will have to follow up on the other thread to see.

**Comment by hexxamillion** [comment]
I think there may be a bug with 2.1.26. I notice for admin settings nothing gets applied after modifying and saving.  I know it's not the db account permissions because it's the same account before upgrade. When I go back to the setting it's like it was not modified. Any idea what would cause this? I am not seeing any errors in the apache log.

**Comment by hexxamillion** [comment]
I found a bug in this. It looks like when the tables are created the table *_misc was not set with the primary key during the upgrade. This explains why any edit I try in the admin settings never applies. In the version 2.1.16 column 'intitule' was set as the primary key.

**Comment by hexxamillion** [comment]
Nevermind..I reinstalled and the table has the primary key set this time. I still cannot modify any of the admin settings. I edit and save then when I go back to the setting it's as if it was never modified.

**Comment by hexxamillion** [comment]
Great news! Password history now shows up. Since I am upgrading from 2.1.16 I found that I had to do sequential upgrade steps in order to successfully upgrade. The issue I do have is the personal items. I get an error stating "Error Encryption of the Password".

**Comment by hexxamillion** [comment]
Here are the steps I needed to take to upgrade from 2.1.16 in case you find it useful.

Due to amount of change I have to upgrade in steps through each version to current.
		○ Start 2.1.22 -> 2.1.23 -> Feature Branch 2.1.24
		
2.1.22 Notes:
	- After upgrading you can't log in using LDAP. 
		○ Error: Field 'field_1' doesn't have a default value
		○ To fix alter table cpassman_log_systems. Uncheck NN for column field_1 and apply. This makes null to be default value.
		○ This fixes login issue.
2.1.23 Notes:
	- After upgrading you can't login using LDAP.
	- To fix copy the identify.php file from 2.1.22 and overwrite the file on current upgrade directory
	- By default items that used to write are now only read. You have to edit the roles and revert them back to write accordingly.

Feature Branch 2.1.24 Notes:
	- When logging in you are prompted to re-encrypt the personal passwords.
	- Error with password encryption for personal items.
	- Password history now works.

**Comment by hexxamillion** [comment]
I went through testing the upgrade again but this time I changed my personal salt key before upgrading from 2.1.16. I made sure it was just numbers and letters. When I got to 2.1.24 and logged in, it went through the re-encryption. I still cannot access the personal items. Same password encryption error. I will try upgrade again but this time wait until I get to version 2.1.23 before changing the personal salt key to see if that makes a difference. Stay tuned..
